bigtable-dataflow-parent/bigtable-hbase-beam/src/main/java/com/google/cloud/bigtable/beam/AbstractCloudBigtableTableDoFn.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.beam;
+
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.hadoop.hbase.client.Connection;
+import org.apache.hadoop.hbase.client.RetriesExhaustedWithDetailsException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.cloud.bigtable.hbase.BigtableConfiguration;
+
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * This class can be used as a superclass for {@link DoFn}s that require a {@link Connection} in
+ * Dataflow.
+ * 
+ * @param <In> the type of the (main) input elements
+ * @param <Out> the type of the (main) output elements
+ */
+public abstract class AbstractCloudBigtableTableDoFn<In, Out> extends DoFn<In, Out> {
+  private static final long serialVersionUID = 1L;
+
+  public static void logRetriesExhaustedWithDetailsException(
```

Comment:
Or maybe move this to a static utility class? The collection of methods in this class feel a bit random...they aren't really related to the DoFn instance